source/common/http/async_client_impl.cc

```diff
@@ -82,11 +83,24 @@ AsyncStreamImpl::AsyncStreamImpl(AsyncClientImpl& parent, AsyncClient::StreamCal
               parent.config_.async_stats_),
       stream_info_(Protocol::Http11, parent.dispatcher().timeSource(), nullptr),
       tracing_config_(Tracing::EgressConfig::get()),
-      route_(std::make_shared<NullRouteImpl>(parent_.cluster_->name(), parent_.singleton_manager_,
-                                             options.timeout, options.hash_policy,
-                                             options.retry_policy)),
       account_(options.account_), buffer_limit_(options.buffer_limit_),
       send_xff_(options.send_xff) {
+  const Router::RetryPolicy* retry_policy = options.parsed_retry_policy;
+  if (options.retry_policy.has_value()) {
```

Comment:
Please move `retry_policy_` initialization into the constructor initializer list. You can do it by using a helper method that returns the `std::unique_ptr<Router::RetryPolicy>`.

If you reorder the `route_` and `retry_policy_` member variables, you can then move route_ initialization into the initializer list as well.

This will allow you to make `retry_policy_` const.

Reply:
done